tests/test_version.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import absolute_import, division, print_function
+
+import pytest
+
+from attr import VersionInfo
+from attr._compat import PY2
+
+
+@pytest.fixture(name="vi")
+def fixture_vi():
+    return VersionInfo(19, 2, 0, "final")
+
+
+class TestVersionInfo:
+    def test_from_string_no_releaselevel(self, vi):
+        """
+        If there is no suffix, the releaselevel becomes "final" by default.
+        """
+        assert vi == VersionInfo._from_version_string("19.2.0")
+
+    @pytest.mark.parametrize("other", [(), (19, 2, 0, "final", "garbage")])
+    def test_wrong_len(self, vi, other):
+        """
+        Comparing with a tuple that has the wrong length raises an error.
+        """
+        assert vi != other
+
+        if not PY2:
```

Comment:
Why is this only py3?

Reply:
Because Python 2 is too YOLO to raise an error if comparison returns `NotImplemented`. :|

However this should be made using a decorator that explains your question.